lib/Controller/FolderController.php

```diff
@@ -150,13 +150,30 @@ public function getFolders(bool $applicable = false, int $offset = 0, ?int $limi
 					return $value;
 				}
 			} else {
-				if (($value = strcmp((string)($a[$orderBy] ?? ''), (string)($b[$orderBy] ?? ''))) !== 0) {
+				if (($value = strnatcmp((string)($a[$orderBy] ?? ''), (string)($b[$orderBy] ?? ''))) !== 0) {
+					// Folder names starting with '_' get pushed to the end while they are brought to the
+					// beginning in the frontend. Do the same here to keep it consistent with the frontend
+					if (strpos($a[$orderBy], '_') === 0 && strpos($b[$orderBy], '_') !== 0) {
+						return -1;
+					}
+
+					if (strpos($b[$orderBy], '_') === 0 && strpos($a[$orderBy], '_') !== 0) {
```

Comment:
To avoid scanning strings:

```suggestion
					if (strncmp($a[$orderBy], '_', 1) === 0 && strncmp( $b[$orderBy], '_', 1 ) !== 0) {
						return -1;
					}

					if (strncmp($b[$orderBy], '_', 1) === 0 && strncmp( $a[$orderBy], '_', 1 ) !== 0) {
```